src/garage/envs/task_onehot_wrapper.py

```diff
@@ -0,0 +1,166 @@
+"""Wrapper for appending one-hot task encodings to individual task envs.
+
+See `~TaskOnehotWrapper.wrap_env_list` for the main way of using this module.
+
+"""
+import akro
+import gym
+import numpy as np
+
+from garage.envs import EnvSpec
+
+
+class TaskOnehotWrapper(gym.Wrapper):
+    """Append a one-hot task representation to an environment.
+
+    See TaskOnehotWrapper.wrap_env_list for the recommended way of creating
+    this class.
+
+    Args:
+        env (gym.Env): The environment to wrap.
+        task_index (int): The index of this task among the tasks.
+        n_total_tasks (int): The number of total tasks.
+
+    """
+
+    def __init__(self, env, task_index, n_total_tasks):
+        assert 0 <= task_index < n_total_tasks
+        super().__init__(env)
+        self._task_index = task_index
+        self._n_total_tasks = n_total_tasks
+        env_lb = self.env.observation_space.low
+        env_ub = self.env.observation_space.high
+        one_hot_ub = np.ones(self._n_total_tasks)
+        one_hot_lb = np.zeros(self._n_total_tasks)
+        self.observation_space = akro.Box(np.concatenate([env_lb, one_hot_lb]),
+                                          np.concatenate([env_ub, one_hot_ub]))
+        self.__spec = EnvSpec(action_space=self.action_space,
+                              observation_space=self.observation_space)
+
+    @property
+    def spec(self):
+        """Return the environment specification.
+
+        Returns:
+            garage.envs.env_spec.EnvSpec: The envionrment specification.
+
+        """
+        return self.__spec
+
+    def reset(self, **kwargs):
+        """Sample new task and call reset on new task env.
+
+        Args:
+            kwargs (dict): Keyword arguments to be passed to env.reset
+
+        Returns:
+            numpy.ndarray: active task one-hot representation + observation
+
+        """
+        return self._obs_with_one_hot(self.env.reset(**kwargs))
+
+    def step(self, action):
+        """gym.Env step for the active task env.
+
+        Args:
+            action (object): object to be passed in env.step(action)
```

Comment:
isn't this known to be a np.ndarray?

Reply:
It's almost always a `np.ndarray`. Theoretically, it can just be an integer for discrete action space environments.